delta-app/src/main/java/io/cdap/delta/app/service/Assessor.java

```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.app.service;
+
+import io.cdap.cdap.api.plugin.InvalidPluginConfigException;
+import io.cdap.cdap.api.plugin.PluginProperties;
+import io.cdap.delta.api.Configurer;
+import io.cdap.delta.api.DeltaSource;
+import io.cdap.delta.api.SourceColumn;
+import io.cdap.delta.api.SourceTable;
+import io.cdap.delta.api.assessment.Assessment;
+import io.cdap.delta.api.assessment.ColumnAssessment;
+import io.cdap.delta.api.assessment.ColumnDetail;
+import io.cdap.delta.api.assessment.ColumnSuggestion;
+import io.cdap.delta.api.assessment.ColumnSupport;
+import io.cdap.delta.api.assessment.PipelineAssessment;
+import io.cdap.delta.api.assessment.Problem;
+import io.cdap.delta.api.assessment.StandardizedTableDetail;
+import io.cdap.delta.api.assessment.TableAssessment;
+import io.cdap.delta.api.assessment.TableAssessor;
+import io.cdap.delta.api.assessment.TableAssessorSupplier;
+import io.cdap.delta.api.assessment.TableDetail;
+import io.cdap.delta.api.assessment.TableList;
+import io.cdap.delta.api.assessment.TableNotFoundException;
+import io.cdap.delta.api.assessment.TableRegistry;
+import io.cdap.delta.api.assessment.TableSummaryAssessment;
+import io.cdap.delta.proto.DBTable;
+import io.cdap.delta.proto.DeltaConfig;
+import io.cdap.delta.proto.FullColumnAssessment;
+import io.cdap.delta.proto.Plugin;
+import io.cdap.delta.proto.Stage;
+import io.cdap.delta.proto.TableAssessmentResponse;
+import io.cdap.delta.proto.TableTransformation;
+import io.cdap.delta.store.DraftNotFoundException;
+import io.cdap.delta.store.InvalidDraftException;
+import io.cdap.transformation.ColumnRenameInfo;
+import io.cdap.transformation.DefaultMutableRowSchema;
+import io.cdap.transformation.DefaultRenameInfo;
+import io.cdap.transformation.TransformationException;
+import io.cdap.transformation.TransformationUtil;
+import io.cdap.transformation.api.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Handles logic around assessment of delta config.
+ */
+public class Assessor {
```

Comment:
Assessment logic which invokes plugins is extracted out of DraftService and moved to a separate class which can also be used by remote tasks. We don't want to use DraftService in remote task as it also makes DB calls.

This is mostly refactored as is with slight change in parameters, refer https://gist.github.com/sumitjnn/1e6e1007536d204920ddaac5fb82f4f8/revisions for comparison with the old code in draft service

Reply:
By DB calls, do you mean calls to the user-provided database, or calls to the CDAP's structured table DB?

Reply:
Its CDAP's structured table DB which stores drafts